routers/leaderboardApi.js

```diff
@@ -0,0 +1,183 @@
+const router = require('express').Router();
+const User = require('../models/users');
+const Question = require('../models/questions');
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
+
+
+// API functionality check
+router.get('/', (req, res, next) => {
+
+	res.render('leaderboard');
+});
+
+//These functions are not needed by the actual game----
+
+// Route to get all question answers
+// router.get('/question/', (req, res, next) => {
+// 	const allQuestions = Question.find({})
+// 		.then(doc => {
+// 			console.log(doc);
+// 			res.status(200).json(doc);
+// 		})
+// 		.catch(err => {
+// 			console.log(error);
+// 			res.status(500).json({
+// 				error: err
+// 			});
+// 		});
+// });
+
+// // Route to create question answers
+// router.post('/question/', (req, res, next) => {
+// 	const question = new Question({
+// 		number: req.body.qno,
+// 		answer: req.body.ans
+// 	});
+
+// 	question
+// 		.save()
+// 		.then(result => console.log(result))
+// 		.catch(err => console.log(err));
+
+// 	res.status(200).json({
+// 		message: "Question object created",
+// 		question: question
+// 	});
+// });
+
+//-----------------------------------------------------
+
+
+// Get sorted list of users by ranking
+router.get('/all', async (req, res, next) => {
+
+	try {
+		const allUsers = await User.find({});
+		allUsers.sort((a, b) => b.questions.length - a.questions.length);
+		const responseJSON = allUsers.map(({uname, questions}) => ({uname, questions}));
+		res.status(200).json(responseJSON);
+	}
+	catch (e) {
+		console.log(e)
+		res.status(500).json({
+			error: e
+		});
+	}
+});
+
+// Get information on a specific user
+router.get('/user', async (req, res, next) => {
+	const uname = req.body.uname?.toString();
```

Comment:
Finally prevented mongoose injection in 1 route.

Reply:
Its there in all routes right?